course_discovery/apps/course_metadata/tests/test_models.py

```diff
@@ -364,6 +365,8 @@ class CourseRunTests(OAuth2Mixin, TestCase):
     @classmethod
     def setUpClass(cls):
         super().setUpClass()
+        call_command('search_index', '--delete', '-f')
+        call_command('search_index', '--create')
```

Comment:
`elasticsearch_dsl_default_connection` unfortunately does not work here. This fixture has a `function` scope, i.e. runs before each test. This behavior is inappropriate because it drops the indexes every time. Here, the test class `CourseRunTests` itself was written so that the previous test uses the context of the next one. This is not very convenient. But this was done. This is why I had to put the command call_command in the `setUpClass` class method that only runs once. This is exactly the behavior we need.

Reply:
So when a test class uses `setUpClass`, it's usually done just for performance reasons (if the tests don't modify the database or whatever, it's easier to do the setup once).

I think if we need to change a test class from class-level setup to function level setup (i.e. change `setUpClass` to `setUp`), I think it's fine. Just a matter of trade offs between performance and maintainability.

Your approach might already be a good trade off there! I don't have a strong opinion. But just wanted to say that if you felt like it would be cleaner to use the fixture, it's alright to retool the test class to accomodate function scope fixtures.

Reply:
Sounds resealable and I have no objections about the fixture usage. So here is the commit https://github.com/raccoongang/course-discovery/pull/12/commits/da31a358c5cc07c79ba048676abc337cf822635d